lib/Doctrine/Common/Proxy/ProxyGenerator.php

```diff
@@ -436,35 +449,35 @@ private function generateMagicGet(ClassMetadata $class)
         $magicGet = <<<EOT
     /**
      * $inheritDoc
-     * @param string \$name
+     * @param string $firstParameterName
      */
-    public function {$returnReference}__get(\$name)
+    public function {$returnReference}__get($parametersBuffer)$returnTypeHint
     {
 
 EOT;
 
         if ( ! empty($lazyPublicProperties)) {
-            $magicGet .= <<<'EOT'
-        if (array_key_exists($name, $this->__getLazyProperties())) {
-            $this->__initializer__ && $this->__initializer__->__invoke($this, '__get', [$name]);
+            $magicGet .= <<<EOT
+        if (array_key_exists($firstParameterName, \$this->__getLazyProperties())) {
+            \$this->__initializer__ && \$this->__initializer__->__invoke(\$this, '__get', [$firstParameterName]);
 
-            return $this->$name;
+            return \$this->$firstParameterName;
         }
 
 
 EOT;
         }
 
         if ($hasParentGet) {
-            $magicGet .= <<<'EOT'
-        $this->__initializer__ && $this->__initializer__->__invoke($this, '__get', [$name]);
+            $magicGet .= <<<EOT
```

Comment:
Please keep NOWDOC and use `%s` + `sprintf()` instead - using escaping and variables inside HEREDOC makes the code even harder to read.